imports/plugins/core/graphql/server/resolvers/account/Mutation/addAccountAddressBookEntry.js

```diff
@@ -1,23 +1,24 @@
-import { decodeAccountOpaqueId, encodeAccountOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/account";
+import { decodeAccountOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/account";
+import { xformAddressResponse } from "@reactioncommerce/reaction-graphql-xforms/address";
 
 /**
  * @name addAccountAddressBookEntry
  * @method
  * @summary resolver for the addAccountAddressBookEntry GraphQL mutation
  * @param {Object} _ - unused
- * @param {Object} args - an object of all arguments that were sent by the client
+ * @param {Object} args.input - an object of all mutation arguments that were sent by the client
+ * @param {String} [args.input.accoundId] - The account ID, which defaults to the viewer account
+ * @param {AddressInput} args.input.address - The address to add
+ * @param {String} [args.input.clientMutationId] - An optional string identifying the mutation call
  * @param {Object} context - an object containing the per-request state
  * @return {Object} AddAccountAddressBookEntryPayload
  */
 export default function addAccountAddressBookEntry(_, { input }, context) {
   const { accountId, address, clientMutationId = null } = input;
-  const dbAccountId = decodeAccountOpaqueId(accountId);
+  const dbAccountId = accountId && decodeAccountOpaqueId(accountId);
```

Comment:
I think we should add at the top of `decodeOpaqueIdForNamespace`:

```js
if (opaqueId === undefined || opaqueId === null) return null;
```

And then we don't need this `accountId &&` check everywhere.

Reply:
👍 Is it possible to go a level deeper and put that check in `decodeOpaqueId`?

Reply:
@ticean `decodeOpaqueId` is not used in `decodeOpaqueIdForNamespace`. They are very identical methods though.

```
export const decodeOpaqueId = (opaqueId) => {
  const unencoded = Buffer.from(opaqueId, "base64").toString("utf8");
  const [namespace, id] = unencoded.split(":");
  return { namespace, id };
};
```

```
export const decodeOpaqueIdForNamespace = curry((namespace, opaqueId, error = new Error(`ID namespace must be ${namespace}`)) => {
  const unencoded = Buffer.from(opaqueId, "base64").toString("utf8");
  const [actualNamespace, id] = unencoded.split(":");
  if (actualNamespace !== namespace) throw error;
  return id;
});
```

Reply:
As per the discussion in stand-up today, if we are going to explicitly require the `accountId` to be passed in, then the check `accountId && decodeAccountOpaqueId(accountId)` is no longer necessary. 

If `opaqueId` is `undefined`, `Buffer.from` will throw an error that could be handled by the client, though it's vague. (Error message example: `"First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object."` )

So...
1. Would you rather return `null` if `opaqueId` is not defined
2. Leave as is and let `Buffer.from` throw the error
3. throw a nicer error if `opaqueId` comes in `undefined`.

Reply:
Why not destructure from a `decodeOpaqueId()` result inside `decodeOpaqueIdForNamespace`? Then you can put the check in `decodeOpaqueId` and keep it nice and DRY.